### ISSUE TITLE
Controller for the content picker field, add a content type parameter

### ISSUE THREAD

**Issue by chinasqzl** [OrchardCMS/OrchardCore#15754]
Contract parts, when you make up two different content types of purchase and sales contracts, the selected partner is different, and you need to read the settings of the contract parts under the content type to extend the IContentPickerResultProvider

**Comment by Piedone** [comment]
Please fill out the issue template (you can get it from [here](https://github.com/OrchardCMS/OrchardCore/issues/new?assignees=&labels=enhancement&projects=&template=feature_request.md&title=)) fully, because I'm unclear on what your issue is about.

**Comment by chinasqzl** [comment]
### Is your feature request related to a problem? Please describe.
public class ContractPart : ContentPart{
  public ContentPickerField Partner { get; set; } = new();
}
public class ContractPartSettings
{
    public string Type { get; set; }  //Sale or Purchase
}

When the contract is a purchase, the partner is the supplier content type, and when it is a sale, the partner is the customer content type, and now ContentPickerAdminController.SearchContentItems() can only get the parts, but it can't get the parts of which content type, and it can't read the ContractPartSettings, so you can't do pickers for more scenarios

![image](https://github.com/OrchardCMS/OrchardCore/assets/16968103/266413e1-b125-4b63-a436-f6376a19d83e)


### Describe the solution you'd like
Contract parts, when you make up two different content types of purchase and sales contracts, the selected partner is different, and you need to read the settings of the contract parts under the content type to extend the IContentPickerResultProvider



### Describe alternatives you've considered
ContentPickerAdminController.SearchContentItems , Add the type parameter
public async Task<IActionResult> SearchContentItems(string type,string part, string field, string query)

**Comment by Piedone** [comment]
I think conceptually this is a similar issue to https://github.com/OrchardCMS/OrchardCore/issues/9444. I this for named/reusable parts?

**Comment by chinasqzl** [comment]
Not a problem, adding a type parameter, mainly for the ContentPickerAdminController.SearchContentItems() you can get the settings in the part, and pass in the IContentPickerResultProvider.Search(), the part is composed of multiple content types, and the settings in the part are not different

**Comment by Piedone** [comment]
Yeah, but isn't this something useful for named/reusable parts? Because `GetPartDefinitionAsync(part).Fields` will be the same as `GetTypeDefinitionAsync(type).Parts.FirstOrDefault(p => p.Name == part).PartDefinition.Fields` except for reusable parts.

**Comment by chinasqzl** [comment]
Yes, then add the parameters of the technical name of the part
